lib/matplotlib/image.py

```diff
@@ -281,7 +282,11 @@ def set_alpha(self, alpha):
         ----------
         alpha : float
         """
-        martist.Artist.set_alpha(self, alpha)
+        if np.isscalar(alpha):
```

Comment:
the docs of np.isscalar basically tell you to not use this function, and suggests np.ndim(alpha) == 0 instead: https://docs.scipy.org/doc/numpy/reference/generated/numpy.isscalar.html

Reply:
also, should this unset any previously set array_alpha?

Reply:
Thanks for the pointer regarding `np.isscalar`. Yes, the `_array_alpha` should also be unset. Having said that, maybe the implementation proposed in https://github.com/tillahoffmann/matplotlib/pull/1 is preferable because it also addresses @jklymak's https://github.com/matplotlib/matplotlib/pull/14889#discussion_r307465796 above.

Reply:
I think I prefer the approach of https://github.com/tillahoffmann/matplotlib/pull/1, yes this means that the type of get_alpha() changed but that seems better than lying as to the alpha value.
OTOH, https://github.com/tillahoffmann/matplotlib/pull/1 also has a change to composite_images and that change appears to not take array_alpha into account?  (Try with two overlapping images that use array alpha, and save to pdf to check.)
It may be fine to just disable software compositing in that case if it's too complicated, and draw the two images separately.